docs/book/src/tasks/certs/auto-rotate-certificates-in-kcp.md

```diff
@@ -0,0 +1,64 @@
+## Automatically rotating certificates using Kubeadm Control Plane provider
+
+When using Kubeadm Control Plane provider (KCP) it is possible to configure automatic certificate rotations. KCP does this by triggering a rollout when the certificates on the machines are about to expire.
+
+Rollout for certificate expiry is automatically enabled for all new clusters created using ClusterAPI v1.3 and above. To enable it for existing clusters see [Enabling Certificate Expiry Rollout for Existing Clusters](#enabling-certificate-expiry-rollout-for-existing-clusters)
+
+### Configuring Machine Rollout
+
+To configure a rollout on the KCP machines you need to set `.rolloutBefore.certificatesExpiryDays` (minimum of 7 days).  
+
+Example: 
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: example-control-plane
+spec:
+  rolloutBefore:
+    certificatesExpiryDays: 21 # trigger a rollout if certificates expire within 21 days
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      ...
+    initConfiguration:
+      ...
+    joinConfiguration:
+      ...
+  machineTemplate:
+    infrastructureRef:
+      ...
+  replicas: 1
+  version: v1.23.3
+``` 
+
+It is strongly recommended to set the `certificatesExpiryDays` to a large enough value so that all the machines will have time to complete rollout well in advance before the certificates expire.
+
+### Triggering Machine Rollout for Certificate Expiry
+
+KCP uses the value in the corresponding Control Plane machine's `Machine.Status.CertificatesExpiryDate` to check if a machine's certificates are going to expire and if it needs to be rolled out.  
+
+`Machine.Status.CertificatesExpiryDate` gets its value from one of the following 2 places:
+
+* `machine.cluster.x-k8s.io/certificates-expiry` annotation value on the Machine object. This annotation is not applied by default and it can be set by users to opt-in in this feature for existing control plane machines (see instructions below).
+* `machine.cluster.x-k8s.io/certificates-expiry` annotation value on the Bootstrap Config object referenced by the machine. This value is automatically set for KCP-managed machines created with CAPI >= v1.3.
```

Comment:
```suggestion
* `machine.cluster.x-k8s.io/certificates-expiry` annotation value on the Bootstrap Config object referenced by the machine. This value is automatically set for machines bootstrapped with CABPK with CAPI >= v1.3.
```

Reply:
if we use bootstrapped with CABPK we should clarify that this apply only to CP machines. (KCP-managed implicitly means both)

Reply:
Hm KCP is only compatible with CABPK, right? Then let's keep the current wording for simplicity